service/history/ndc/workflow_state_replicator.go

```diff
@@ -488,7 +490,11 @@ func (r *WorkflowStateReplicatorImpl) applySnapshot(
 	if err != nil {
 		return err
 	}
-	// Todo: localMutableState.ApplySnapshot
+
+	err = localMutableState.ApplySnapshot(snapshot)
+	if err != nil {
+		return err
+	}
```

Comment:
We need to call ms.PopTasks() to clear all generated tasks during ms.ApplySnapshot().

Reply:
Added.

Reply:
syncSubStateMachinesByType uses node.Sync() which internally performs a task refresh for the synced node. ([comment](https://github.com/temporalio/temporal/pull/6517/files#r1769743110))